### PR TITLE
copyCell Grid Context Menu Action (Fixes #690)

### LINKED ISSUE #690: Enhance Copy Support in grid
Would like to have a build-in ability to copy a *cell* value only  as well as the entire selection.
This is a facility we had in Hoist-Sencha.

Need to make sure our RecordActions are getting metadata for column/cell to do this.

### PR DESCRIPTION
**Pull request by haynesjm42**
Support for new copyCell action which works off the record and column that was clicked on. Uses https://www.npmjs.com/package/clipboard-copy for putting the data into the clipboard programmatically.

Also added changes to use our standard copy icon for the built-in copy actions.

Note that this branch originally contained additional changes to use our own implementation for serializing the grid rows for copying to allow for more flexibility than what ag-grid provides, have preserved that and will explore further in the future, but wanted this feature to be able to get in separately since it has come up multiple times.

Hoist P/R Checklist
-------------------
Review and check off the below. Items that do not apply should be checked to indicate they have been considered. If unclear if a step is relevant, please leave unchecked and note in comments.

- [x] Up to date with `develop` branch as of last change.
- [x] Ready for review (or add `wip` label).
- [x] Added CHANGELOG entry (or N/A)
- [x] Checked for breaking changes (add `breaking-change` label + CHANGELOG if so, or N/A)
- [x] Updated doc comments / prop-types (or N/A)
- [x] Reviewed and tested on Mobile (or N/A)
- [x] Created Toolbox branch / PR (link provided here, or N/A)

### CODE REVIEW

File: kit/clipboard-copy/index.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+/*
+ * This file belongs to Hoist, an application development toolkit
+ * developed by Extremely Heavy Industries (www.xh.io | info@xh.io)
+ *
+ * Copyright © 2019 Extremely Heavy Industries Inc.
+ */
+
+import copy from 'clipboard-copy';
+
+export {copy as copyToClipboard};
```
Comment (lbwexler) [inline kit/clipboard-copy/index.js]: John -- why do we need this?  Would it be possible to do this with the 'clipboard' package that we already have.....

Comment (haynesjm42) [reply kit/clipboard-copy/index.js]: I looked at clipboard.js and didn't see any way to trigger the copy to clipboard programmatically. It looked like it always needed to be attached to an element in order to function, which would be a little tricky with what ag-Grid provides us, and would generally require a lot more wiring to function if it is even possible. Unless I am missing something it definitely would not be as simple as the one-liner that the clipboard-copy package provides.

**Comment by lbwexler** [comment]
Could we easily simplify the few places where we are using 'clipboard'?
Uncomfortable with having two libraries that purport to provide the same
exact functionality.  Is it possible you just found something better?

**Comment by haynesjm42** [comment]
Will look into it a bit more to see if we can/should replace clipboard.js with clipboard-copy.

**Comment by haynesjm42** [comment]
So the only usage we have of clipboard.js is the ClipboardButton component. We could replicate the functionality there using the copy-clipboard library but it would be more complicated, as clipboard.js is perfect for that use-case. clipboard.js is excellent if you have an element which you want to trigger copying (or cutting) of another elements content, while clipboard-copy is excellent if you just want to put some arbitrary data into the clipboard. They both serve their use-case very well, but not so well for the other use-case, so they don't really provide the same functionality and I think it makes sense to include both.

**Comment by lbwexler** [comment]
I was able to update the existing classes to use the new promise based library.  I always thought the existing implementation was way too confusing anyway!

Also, please note that I dropped the 'kit' entry.  Seems like overkill here to me -- unless we are installing element factories, or somehow adding value, not sure what the point of a 'kit' entry is....

Let me know your thoughts -- if you are fine with it, ok to merge

**Comment by haynesjm42** [comment]
Looks good to me, and thanks for taking care of eliminating the other library, looks like a good simplification to the ClipboardButton interface. And agree the kit was overkill, not entirely sure what my reasoning was for adding that :)

File: package.json [hunk 2]
```diff
@@ -24,6 +24,7 @@
     "ag-grid-react": "~21.1.0",
     "classnames": "~2.2.6",
     "clipboard": "~2.0.4",
+    "clipboard-copy": "^3.0.0",
```
Comment (amcclain) [inline package.json]: @lbwexler / @haynesjm42  - saw the thread re. avoiding the two libraries, but doesn't look like the old one was actually removed... 😀OK to actually take it out?

Comment (lbwexler) [reply package.json]: great catch on this one.  Removed!

**Pull request merged**